### ISSUE TITLE
Create a new compiler script 

### ISSUE THREAD

**Issue by Cyrof** [KShiMin/Nursery-Tic-Tac-Toe#4]
## Need for new compiler script
The current compiler script is created using shebang script which is fine but it works differnetly in different OS. Therefore, there is a need to create a universal compiler script for all OSes. 

## Use Case
The new compiler script should be universal and be used the same in every OS. 

## Language 
The new compiler should use a language that is compatible on all platforms such as `python`.

## Progress Status
- [x] Code Commited
- [x] Testing - QA
- [x] Documentation (README file - resolved isseus, compatibility, known issues)

**Comment by Cyrof** [comment]
**Commited Codes Updates**
The codes have been commited in e3cd7684ba5f766b30edfade23373675f6e5503e.
QA testing will be done soon.

**Comment by Cyrof** [comment]
**Release Update** 
Created github actions to automate creation of executable for different OS. The github action will only be trigger when a tag is created and push to repository. Executables created from by github actions will be upload as artifacts and will be manually downloaded and uploaded in new release based on the new tag.

**First Release**
The first release is based on `v0.1.0` tag and includes the first implementation of executable.

**Comment by KShiMin** [comment]
Hi @Cyrof , I have tried the executable file for windows and encountered errors as shown below. 

```
PS C:\Programming\Nursey-Tic-Tac-Toe> .\compile.exe -f test.c -o test                    
Traceback (most recent call last):
  File "compile.py", line 135, in <module>
  File "compile.py", line 128, in main
  File "compile.py", line 50, in compile_file
  File "compile.py", line 41, in _normalise_path
  File "subprocess.py", line 548, in run
  File "subprocess.py", line 1026, in __init__
  File "subprocess.py", line 1538, in _execute_child
FileNotFoundError: [WinError 2] The system cannot find the file specified
[PYI-17048:ERROR] Failed to execute script 'compile' due to unhandled exception!
```

Here is my folder structure:
```
│   .gitignore
│   compile.exe
│   LICENSE
│   README.md
│   test.c
│
├───.github
│   └───workflows
│           build.yaml
│
├───compiler-tools
│       compile.py
│
└───tic-tac-toe
        game_logic.c
        game_logic.h
        player_vs_cpu.c
        player_vs_cpu.h
        player_vs_player.c
        player_vs_player.h
        tic_tac_toe.c
```

**Comment by Cyrof** [comment]
Thanks for reporting this issue @KShiMin. I will take a look at this issue when i have the time.

## Important
For future comments on issues **do not** add screenshots but instead copy and paste the output into the comment and place it like this 
```
code here 
```

If possible do try to edit your comment to include the error code as well as your folder structure instead of the screenshot.

**Comment by Cyrof** [comment]
## Update 
- Resolved the `FileNotFoundError` issues encountered on Windows. 
- Created a new release under the tag `v0.1.1`.
- Initial functionality checks look good, and I am now moving on to thorough quality testing to ensure everything works as expected.

**Comment by Cyrof** [comment]
## Update
- Completed thorough QA testing, and everything looks good to me.
- The `FileNotFoundError` issue on Windows is resolved, and the functionality works as expected across platforms. 
- The release tagged `v0.1.1` is ready for use.

**Comment by Cyrof** [comment]
## Issue Resolved
The issue with the new auto compiler has been resolved here 2ef1686cb178b2f5ded2d9b18de5e24ff5585fba. The documentation in the `README` has also been updated to reflect the changes and provide clear instructions for using the compiler here 8dff3dfc5c68d0c84e8df33eaa653a55bb674b78.  

This issue will now be closed. If there are any further problems or improvements needed, feel free to open a new issue. 

Thank you for your patience and feedback!
